### PR TITLE
Add EMIS ons_deaths and ons_deaths_raw tables

### PR DESCRIPTION
**Pull request by rebkwok**
Part of #797 

This adds the ons_deaths and ons_deaths_raw tables for EMIS.  It also required some changes to the TPP tables, because ons_deaths was previously defined in `tables.beta.core` and `tables.beta.raw.core` and included the place of death column, which is only available in TPP.

Note that the "raw" EMIS table isn't really very raw; in EMIS the ONS table doesn't have a reference to the patient id (registration_id) and so requires a join on the patient table, which itself needs deduplication. In addition, in EMIS the ONS table contains every upload, so it's filtered to the last upload date - and I think we still need to do this if we're keeping the "raw" version of the table similar to the raw version in TPP and the description in the core table (i.e. the table is the same as the beta.ons_deaths one, but also includes duplicate deaths)

### CODE REVIEW

File: ehrql/tables/beta/raw/core.py [hunk 1]
```diff
@@ -36,8 +35,7 @@ class ons_deaths_raw(EventFrame):
     - [User guide to mortality statistics](https://www.ons.gov.uk/peoplepopulationandcommunity/birthsdeathsandmarriages/deaths/methodologies/userguidetomortalitystatisticsjuly2017)
     - [How death registrations are recorded and stored by ONS](https://www.ons.gov.uk/aboutus/transparencyandgovernance/freedomofinformationfoi/howdeathregistrationsarerecordedandstoredbyons)
 
-    In the associated database table [ONS_Deaths](https://reports.opensafely.org/reports/opensafely-tpp-database-schema/#ONS_Deaths),
-    a small number of patients have multiple registered deaths.
+    In the associated database table a small number of patients have multiple registered deaths.
```
[the remark this answers is not in the document]
Comment (rebkwok) [reply ehrql/tables/beta/raw/core.py:38]: I sort-of think we can - shouldn't the ONS deaths table be the same in emis and TPP? It comes from the same source (at least, I believe it does).  For the same reason, I think we do want a raw version, if we're including it for TPP on the basis that users might need to do some sort of checks on duplicate death rows.

[the remark this answers is not in the document]
Comment (iaindillingham) [reply ehrql/tables/beta/raw/core.py:38]: I guess my thinking is that although the data come from the same source, TPP may have a different ETL process than EMIS.

[the remark this answers is not in the document]
Comment (rebkwok) [reply ehrql/tables/beta/raw/core.py:38]: But whatever the ETL process, both underlying tables have a small number of patients with multiple registered deaths (I checked), and in the non-raw table we handle removing those in the same way, so I think this statement is still relevant.

[the remark this answers is not in the document]
Comment (iaindillingham) [reply ehrql/tables/beta/raw/core.py:38]: Ah, okay, if we know because you've checked, then we _can_ say this. Thank you for clarifying.

**Pull request merged**